Fix TIME_T_MAX_BITS on 32-bit systems with a 32-bit signed time_t.
https://github.com/dovecot/core/pull/316

diff --git a/m4/gmtime_max.m4 b/m4/gmtime_max.m4
index d2501c1776..a0d8e06780 100644
--- a/m4/gmtime_max.m4
+++ b/m4/gmtime_max.m4
@@ -21,12 +21,10 @@ AC_DEFUN([DOVECOT_GMTIME_MAX], [
              Let's just do the same as Cyrus folks and limit it to 40 bits. */
           bits = 40;
         }
-        #ifdef TIME_T_SIGNED
         if (bits == 32) {
           /* Signed 32-bit time_t is essentially the same as unsigned 31-bit time_t */
           bits = 31;
         }
-        #endif
     
         f = fopen("conftest.temp", "w");
         if (f == NULL) {
